### PR TITLE
Add getting started guide for Kubernetes

### PR DESCRIPTION
**Pull request by moutoum**
### What does this PR do?

This PR adds a new getting started guide for Kubernetes using native ingresses.

### Motivation

The documentation needs more beginner oriented contents.

### CODE REVIEW

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (ddtmachado) [inline docs/content/getting-started/quick-start-with-kubernetes.md:11]: This was probably not your intention but you agree this literally says that Traefik is difficult to use on Kubernetes?
I suggest rephrasing it so we don't scare newcomers right in the beginning 😂

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (ddtmachado) [inline docs/content/getting-started/quick-start-with-kubernetes.md:72]:
```suggestion
And then, it's time to bind the role on the account to apply the
```
wdyt? to avoid calling it "last operation" as it is the last only for the RBAC control context

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (ddtmachado) [inline docs/content/getting-started/quick-start-with-kubernetes.md:101]: This doesn't really explain what is an ingress controller.
WYDT of taking inspiration from or even linking directly to our glossary page?
https://traefik.io/glossary/kubernetes-ingress-and-ingress-controller-101/

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (ddtmachado) [inline docs/content/getting-started/quick-start-with-kubernetes.md:262]: Now it's expands on the concept of an ingress controller that was left in the air previously. If we clarify this there, on the beginning, I believe this section can be removed.

**Review by ddtmachado — COMMENTED** [review]
Thanks for the guide! I left a few non technical comments for consideration

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:62]:
```suggestion

!!! info "The reference for this file can be found [there](../../reference/dynamic-configuration/kubernetes-crd/#rbac)."

```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:97]:
```suggestion
!!! info "`roleRef` is the Kubernetes reference to the role created in `00-role.yml`."

!!! info "`subjects` is the list of accounts reference."

    In the guide, it only contains the account created in `00-account.yml`
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:103]:
```suggestion
To start Traefik on the Kubernetes cluster,
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:100]:
```suggestion

!!! info "This section can be managed with the help of the [Traefik helm chart](../install-traefik/#use-the-helm-chart)."

```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:147]:
```suggestion
In this deployment,
the static configuration enables the Traefik dashboard
and uses Kubernetes native ingress resources as router definitions to route incoming requests.
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:144]:
```suggestion
From here, it is possible to enable the dashboard,
configure entry points,
select dynamic configuration providers,
and [more](../reference/static-configuration/cli.md)...
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:152]:
```suggestion
!!! info "When there is no entry points in the static configuration"

    Traefik creates a default one called `web` using the port `80` routing HTTP requests.

!!! info "When enabling the [`api.insecure`](../../operations/api/#insecure) mode, Traefik exposes the dashboard on the port `8080`."
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:157]:
```suggestion
A deployment manages scaling and then can create lots of containers, called [Pods](https://kubernetes.io/docs/concepts/workloads/pods/).
Each pod is configured following the `spec` field in the deployment.
Given that it can exist a lot of Traefik instances,
a piece is required to forward the traffic to any of the instance:
a [`Service`](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#Service).
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:158]: Which process?

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:191]:
```suggestion
!!! warning "It is possible to expose a service in different manners."

    Depending on your working environment and use case,
    the `spec.type` might change.
    It is strongly recommended to understand the available [service types](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types).
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:198]:
```suggestion
It is now time to apply those files on your cluster to start Traefik.
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:212]:
```suggestion
The now missing part is the business application behind the reverse proxy.
For this guide, [traefik/whoami](https://github.com/traefik/whoami) is used,
but the concepts are equivalent to any other application.
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:216]:
```suggestion
The `whoami` application is a simple HTTP server running on port 80 which answers host-related information to the incoming requests.
As usual, start by creating a file called `03-whoami.yml` and
paste the following `Deployment` resource:
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:264]:
```suggestion
Thanks to the Kubernetes API,
Traefik is notified when an Ingress resource is created, updated, or deleted.
It makes the process very dynamic.
The ingresses are, in a way, the [dynamic configuration](../../providers/kubernetes-ingress/) for Traefik.
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:292]:
```suggestion
At this point, all the configurations are ready.
It is time to apply those new files:
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:301]:
```suggestion
If everything is applied correctly, it should be possible to access the `whoami` application and the Traefik dashboard.
Try to load the dashboard in a web browser: `http://<external-ip>:8080`.
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:303]:
```suggestion
Then try to access the `whoami` application:
```

File: docs/mkdocs.yml
Comment (tomMoulard) [inline docs/mkdocs.yml:70]:
```suggestion
      - 'Quick Start':
          - 'Docker': 'getting-started/quick-start.md'
          - 'Kubernetes': 'getting-started/quick-start-with-kubernetes.md'
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:306]:
```suggestion
              -f 04-whoami-ingress.yml
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:136]:
```suggestion
          image: traefik:v2.8
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (tomMoulard) [inline docs/content/getting-started/quick-start-with-kubernetes.md:315]:
```suggestion
curl -v http://localhost/
```

**Comment by tomMoulard** [comment]
We are waiting on @EskDhg review to go forward to validate this PR.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:2]: No need to add "quickly" in the title. A getting started guide implies "quickly"

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:3]: Same comment as above about "quickly"

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:8]: A Simple Use Case of Traefik Proxy and Kubernetes

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:11]: This guide is an introduction to using Traefik Proxy...

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:12]: The objective is to learn how to...

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:19]: In order to use the Kubernetes API,

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:23]: in this case, Traefik Proxy.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:66]: You can find the reference for this file [here]

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:105]: Helm chart, Helm needs to be capitalized

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:165]: Pod is capitalized in the previous sentence. Choose the appropriate capitalization and be consistent throughout the article.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:166]: This sentence doesn't make sense. Do you mean "Given that pods can exist in a lot of Traefik instances"?

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:167]: a location?

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:168]: ... to forward the traffic to any of the instances, namely a Service.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:199]: in different ways

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:203]: ... understand the available service types before proceeding to the next step.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:217]: The only part still missing is the business application behind the reverse proxy.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:218]: For this guide, we use the example application [traefik/whoaimi,

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:219]: but the principles are applicable to any other application.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:271]: This makes the process dynamic.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:276]: ... in the official Kubernetes documentation.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:309]: Now you should be able to access ....

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [reply docs/content/getting-started/quick-start-with-kubernetes.md:310]: Load the dashboard on a web browser: ...

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (EskDhg) [inline docs/content/getting-started/quick-start-with-kubernetes.md:312]: And now access the 'whoami' application:

**Review by EskDhg — DISMISSED** [review]
Hi, Maxence,

I reviewed the guide and left some comments. Let me know if everything is clear to you. If not, ping me on slack.

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (moutoum) [reply docs/content/getting-started/quick-start-with-kubernetes.md:165]:
```suggestion
Each Pod is configured following the `spec` field in the deployment.
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (moutoum) [reply docs/content/getting-started/quick-start-with-kubernetes.md:166]:
```suggestion
Given that a Deployment can run multiple Traefik Proxy Pods,
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (moutoum) [inline docs/content/getting-started/quick-start-with-kubernetes.md:168]:
```suggestion
a piece is required to forward the traffic to any of the instance:
namely a [`Service`](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#Service).
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (mloiseleur) [inline docs/content/getting-started/quick-start-with-kubernetes.md:68]:
```suggestion
The next step is to create a dedicated service account for Traefik.
```

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (mloiseleur) [inline docs/content/getting-started/quick-start-with-kubernetes.md:69]:
```suggestion
In a file called `00-account.yml`, put the following [`ServiceAccount`](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/service-account-v1/#ServiceAccount) resource:
```

**Review by kevinpollet — APPROVED** [review]

**Review by rtribotte — APPROVED** [review]

File: docs/content/getting-started/quick-start-with-kubernetes.md
Comment (mpl) [inline docs/content/getting-started/quick-start-with-kubernetes.md:154]:
```suggestion
!!! info "When there is no entry point in the static configuration"
```

**Review by mpl — CHANGES_REQUESTED** [review]
LGTM after typo fix

**Review by mpl — APPROVED** [review]

**Pull request merged**
